maintainer-guide.md

```diff
@@ -0,0 +1,79 @@
+# Maintainer's guide
```

Comment:
Is this for Maintainers or Contributors? Some of the suggestions seem oriented toward anyone who might contribute, vs. just those with commit access. I've seen a convention of `MAINTAINERS.md` and `CONTRIBUTORS.md` for these.